utils/src/command.rs

```diff
@@ -33,6 +33,11 @@ pub trait CommandRunExt {
     /// Execute the child process, parsing its stdout as JSON. This uses `run` internally
     /// and will return an error if the child process exits abnormally.
     fn run_and_parse_json<T: serde::de::DeserializeOwned>(&mut self) -> Result<T>;
+
+    /// Check if the command can be executed by running it and ignoring its output. Returns `true`
+    /// if the command exists, `false` otherwise. The exit code is ignored. Only run this with
+    /// commands that have no undesired side effects, e.g. `dnf --version`.
+    fn can_execute(&mut self) -> bool;
```

Comment:
I'd prefer searching `$PATH` and looking for the executable vs running it as a no-op.

Reply:
If we do that we won't have stuff like:

- Checking that it's executable
- Checking that we have permissions
- Checking that the architecture is right
- Checking that we have all the shared libraries

Also searching the $PATH sounds hard? From quick googling it looks like our best bet is the `which` crate

(I don't mind doing it, just want to understand the motivation/benefits)

Reply:
At a quick glance the `which` crate looks sane, and [its reverse dependencies](https://crates.io/crates/which/reverse_dependencies) include some important crates.

> Checking that it's executable

This one is trivial to check.

> Checking that we have permissions

What permissions are there other than the executable bit? OK if one starts to get into real details there are e.g. ACLs but I don't think we need to care about that

> Checking that the architecture is right
> Checking that we have all the shared libraries

If one of these is wrong the system is just broken and it's not our fault; I mean how would we deal with a `/usr/bin/podman` with the wrong architecture? It's not likely a `dnf install` would fix that if it somehow happened anyways...